File: backend/app/schemas/album.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel


class AlbumBase(BaseModel):
    title: str


class AlbumCreate(BaseModel):
    title: str
    creator: str
    song_ids: List[int]


class AlbumUpdate(BaseModel):
    title: Optional[str] = None
    creator: Optional[str] = None
    song_ids: Optional[List[int]] = None


class AlbumRead(BaseModel):
    id: int
    title: Optional[str] = None
    creator: Optional[str] = None
    song_ids: Optional[List[int]] = None
    user_ids: Optional[List[int]] = None

    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/types/album.ts
export type Album = {
  id: number
  title: string | null
  creator: string | null
  song_ids: number[] | null
  user_ids: number[] | null
  created_at: string
  updated_at: string
}

export type AlbumCreate = {
  title: string
  creator: string
  song_ids: number[]
}

export type AlbumUpdate = {
  title?: string | null
  creator?: string | null
  song_ids?: number[] | null
}

